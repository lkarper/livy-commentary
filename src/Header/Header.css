.Header__header {
    position: fixed;
    height: 100px;
    width: 100%;
    background-color: #FFFFF0;
    color: black;
    z-index: 100;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    padding: 0 20px;
    box-shadow: 0px 1px 5px 0px #B2AEB2;
}

.Header__link, .Header__link:visited {
    font-size: 20px;
    color: black;
}

.Header__h1-a, .Header__h1-a:visited, .Header__h1-a:active {
    color: black;
    text-decoration: none;
}

.Header__h1-a:hover, .Header__h1-a:focus {
    text-decoration: underline;
}

.Header__h1 {
    font-size: 20px;
}

@media screen and (min-width: 750px) {
    .Header__h1 {
        font-size: 32px;
    }
}