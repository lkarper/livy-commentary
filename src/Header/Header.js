import React, { useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import TokenService from '../services/token-service';
import IdleService from '../services/idle-service';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import MainNav from './MainNav/MainNav';
import './Header.css';

const Header = (props) => {

    const { forceUpdate } = props;

    const [showNav, toggleShowNav] = useState(false);

    const handleLogout = () => {
        TokenService.clearAuthToken();
        TokenService.clearCallbackBeforeExpiry();
        IdleService.unRegisterIdleResets();
        forceUpdate();
    }

    return (
        <>
            <header className={`Header__header ${showNav ? 'show-nav' : ''}`}>
                <div
                    className='Header__top-container'
                >
                    <h1 className='Header__h1'>
                        <Link
                            to='/'
                            className='Header__h1-a'
                        >
                            The Third Decade Project
                        </Link>
                    </h1>
                    {TokenService.hasAuthToken() &&
                        <Link
                            to='/'
                            className='Header__link'
                            onClick={handleLogout}
                        >
                            Logout
                        </Link>
                    }
                    <button
                        className='Header__mobile-button'
                        onClick={() => toggleShowNav(!showNav)}
                    >
                        <FontAwesomeIcon
                            className='Header__burger' 
                            icon={faBars}
                        />
                    </button>
                    <div
                        className='Header__widescreen-nav-container'
                    >
                        <MainNav />
                    </div>
                </div>
                <div
                    className={`Header__mobile-nav-container ${showNav ? 'show' : 'hide'}`}
                >
                    <MainNav 
                        toggleShowNav={toggleShowNav}
                    />
                </div>
            </header>
        </>
    );
}

Header.defaultProps = {
    forceUpdate: () => {},
};

Header.propTypes = {
    forceUpdate: PropTypes.func,
};

export default Header;
