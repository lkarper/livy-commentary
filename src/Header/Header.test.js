import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from './Header';

describe('Header component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
