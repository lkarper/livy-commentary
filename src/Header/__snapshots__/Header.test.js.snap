// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header
      forceUpdate={[Function]}
    >
      <header
        className="Header__header"
      >
        <h1>
          <Link
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                Commentarius de bello Punico
              </a>
            </LinkAnchor>
          </Link>
        </h1>
      </header>
    </Header>
  </Router>
</BrowserRouter>
`;
